### DISCUSSION TITLE
Group by In Prisma

### DISCUSSION THREAD

**Discussion by EasyTrust** [prisma/prisma#23391] [Q&A] [answered]
### Question

I Have two Prisma Model Order and Order Process Type, I want to count the Order base on order process type i want the Result Like this
```
[
    {
        "_count": {
            "_all": 2
        },
        "orderProcessTypeId": 1,
        "processName": "In Progress"
    },
    {
        "_count": {
            "_all": 1
        },
        "orderProcessTypeId": 2,
         "processName": "Completed"
    }
]
```
I Tried Like this
```
const data = await prisma.order.groupBy({
      by: ['orderProcessTypeId'],
      _count: {
        _all: true,
      },
    
    });
```
But in the Result i get only orderProcessTypeId But i want the Process Type name also

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model Order {
  orderId            Int              @id @default(autoincrement())
  orderDate          DateTime         @default(now())
  orderProcessTypeId Int
  orderProcessType   OrderProcessType @relation(fields: [orderProcessTypeId], references: [orderProcessTypeId])
  total              Float
}

model OrderProcessType {
  orderProcessTypeId Int    @id @default(autoincrement())
  processName       String @db.VarChar(50)
  orders              Order[]

}
```

- OS: Windows
- Database: 
- Node.js version: 20.10.0

- prisma v 5.3.1

**Comment by ludralph** [comment, marked as answer]
Hi @EasyTrust 👋 

Thank you for raising this question.

Currently `groupBy` doesn't permit the inclusion of related model fields. There is an open issue to address this.

- https://github.com/prisma/prisma/issues/16243

However you can use this code below as a workaround:
```js
const orderByProcessTypeId = await prisma.order.groupBy({
    by: ['orderProcessTypeId'],
    _count: {
      _all: true,
    },
  });
  
  const result = await Promise.all(orderByProcessTypeId.map(async (item) => {
    const processType = await prisma.orderProcessType.findUnique({
      where: { orderProcessTypeId: item.orderProcessTypeId },
    });
    return {
      ...item,
      processName: processType.processName,
    };
  }));
```

In this code, result will be an array of objects, each containing `_count`, `orderProcessTypeId`, and `processName`.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
